### DISCUSSION TITLE
How to serve files directly?

### DISCUSSION THREAD

**Discussion by tina-junold** [directus/directus#16137] [Archive]
Hi,

while i really like Directus and in addition GraphQL saves a lot of time (and requests), we have still a problem to serve files (like images, PDF files, ...) directly. While it's possible to add the `access_token` to the URL this solution is *ugly*. Loading all files with a subroutine in JavaScript and inject pictures in the DOM doesn't seem practical ether.
Is there another way to access the files directly? Since we use an custom S3 it's not possible to access  the files from filesystem. And setting the permission to public is working but a security risk.

We tried to set the access token to an cookie, like it's possible with the [refresh token](https://docs.directus.io/reference/authentication.html#login).

**Comment by br41nslug** [comment]
If you want to keep the authentication in place you'll need a way to provide the access token. Directly in the URL may not be the prettiest solution it is the most versatile. Alternatively you can provide the token as header but that's not really supported in `<img>` tags.

**Reply by rijkvanzanten** [reply]
Or make the file read access public 🙂

**Comment by tina-junold** [comment]
It would be nice if a Directus would support a cookie based solution. Maybe this could be a feature....
